### PR TITLE
add reference model logps to chunkedloss interface and fix dpo loss fn

### PR DESCRIPTION
**Pull request by shivam15s**
accomodate reference model logps in chunked loss interface and make dpo loss use reference model logps in its loss function
## Summary

as title

## Testing Done

- Hardware Type: <BLANK>
- [ ] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: src/liger_kernel/chunked_loss/fused_linear_preference.py [hunk 1]
```diff
@@ -18,6 +18,30 @@ def preference_loss_fn(chosen_logps, rejected_logps, beta=0.1):
         """
         raise NotImplementedError("Preference loss function must be implemented.")
 
+    @staticmethod
+    def get_ref_logps(
+        input_chunk, ref_weight, target_chunk, ref_bias=None, ignore_index=-100
+    ):
+        with torch.no_grad():
+            ref_logits_chunk = input_chunk @ ref_weight.t()
```
Comment (pramodith) [inline src/liger_kernel/chunked_loss/fused_linear_preference.py:26]: The logic to get the log_probs is common for both the reference and active policy model, wondering if we can create a util function that can be called by both instead of repeating the code.

**Review by pramodith — COMMENTED** [review]
Great work! Wrapping DPO up.

**Review by pramodith — APPROVED** [review]
Thanks for the refactor!
